src/sql/transform.rs

```diff
@@ -54,11 +69,7 @@ impl AggFuncRewriter {
             over: None,
             distinct,
         });
-        Expr::BinaryOp {
-            left: Box::new(sum),
-            op: BinaryOperator::Divide,
-            right: Box::new(count),
-        }
+        Self::plan_divide(sum, count)
```

Comment:
Are there cases where we would expect a zero count and a non-zero sum? I'm sure we can work ourselves in to such a place, but .. is there one you are thinking of?

Reply:
Relatedly, the new division planning is .. SQL behavior preserving? Should we be producing a runtime error, or producing NULL instead? If I had to guess I would say this was a change in the behavior, and some notes about what the goal is would help.

Reply:
Yeah, it's SQL behavior preserving. Aggregate functions never return division by zero errors, but we lift the division out of the implementation of the aggregate and into a normal scalar expr. This was previously fine because our scalar expr division would convert division by zero errors into NULLs, but now it might return division-by-zero errors, so we need to explicitly check for and suppress those.